CMakeLists.txt

```diff
@@ -398,12 +398,30 @@ endforeach (TARGET ${JUCE_TARGET_LIST})
 # ============================
 foreach (TARGET ${JUCE_TARGET_LIST})
 if (NOT MSVC)
-	target_compile_options (${TARGET} PRIVATE
-			-pthread
-			-fvisibility=hidden
-			-fvisibility-inlines-hidden
-			"-Wa,-mbig-obj"
-	)
+	if (CMAKE_COMPILER_IS_GNUCXX)
+		include(CheckCXXCompilerFlag)
+		check_cxx_compiler_flag("-Wa,-mbig-obj" HAS_BIG_OBJ_FLAG)
+		if (HAS_BIG_OBJ_FLAG)
+			target_compile_options (${TARGET} PRIVATE
+					-pthread
```

Comment:
These three shared compilation options can be merged together, and then the big obj option can be judged separately using the if condition